hono/src/services/items.ts

```diff
@@ -23,8 +23,50 @@ const selectItemSchema = createSelectSchema(items, {
   id: z.string().refine((val) => isCuid(val)),
 })
 
-app.get('/', injectDB, async (c) => {
-  return c.json(await c.get('db').query.items.findMany())
+const paginationValidationItems = z.object({
+  page: z
+    .string()
+    .refine((val) => !isNaN(+val) && +val >= 0, {
+      message: 'Items page number must be a non-negative number',
+    })
+    .optional(),
+  size: z
+    .string()
+    .refine((val) => !isNaN(+val) && +val > 0, {
+      message: 'Items page size must be a positive number',
+    })
+    .optional(),
```

Comment:
Is there a reason we can't just use `z.number().nonnegative().safe()` for page, and `z.number().positive().safe()` for size? Also, I think it would be wise to set an upper bound on the max size, say 1000 entries? 

[Zod docs for numbers](https://zod.dev/?id=numbers)

Reply:
Yeah, it looks like `page` and `size` are expected to be strings and I cannot cast them to numbers. I can implement an upper bound.

Reply:
Are you sure? I believe this should work, see https://github.com/radicalplatforms/shafa/pull/110/files/12fffa280a6254d2087804ea37434a92a23ee073#diff-a9e37437a7f0a721da7c6f523cfcdef37c5605aaae739da9c365bb881fa795dcR18

Reply:
Oooh, I see why it is doing that. What about coercion https://zod.dev/?id=coercion-for-primitives?

Reply:
Both approaches ended up not working, falling back to original solution.